libcalico-go/lib/backend/model/block_affinity.go

```diff
@@ -50,11 +63,12 @@ type BlockAffinity struct {
 }
 
 func (key BlockAffinityKey) defaultPath() (string, error) {
-	if key.CIDR.IP == nil || key.Host == "" {
+	if key.CIDR.IP == nil || key.AffinityCfg.Host == "" {
```

Comment:
This needs to assert on the `key.AffinityCfg.AffinityType` being missing as well.